clippy_lints/src/types.rs

```diff
@@ -267,6 +291,27 @@ impl Types {
                 let res = qpath_res(cx, qpath, hir_id);
                 if let Some(def_id) = res.opt_def_id() {
                     if Some(def_id) == cx.tcx.lang_items().owned_box() {
+                        if_chain! {
+                            let last = last_path_segment(qpath);
+                            if let Some(ref params) = last.args;
+                            if !params.parenthesized;
+                            if let Some(ty) = params.args.iter().find_map(|arg| match arg {
+                                GenericArg::Type(ty) => Some(ty),
+                                _ => None,
+                            });
+                            if let TyKind::Rptr(..) = ty.kind;
+                            let ty_ty = hir_ty_to_ty(cx.tcx, ty);
+                            then {
+                                span_lint_and_help(
+                                    cx,
+                                    BOX_BORROWS,
+                                    hir_ty.span,
+                                    "usage of `Box<&T>`",
```

Comment:
We should use `span_lint_and_suggestion` here and emit a suggestion instead of a help

Reply:
With the text being something like "`Box<&T>` does not offer any advantages over `&T`, please try: {}"

Reply:
You'd want to also add a fixture test for this (add run-rustfix to the top and run/update the tests a couple times)